nav2_dwb_controller/dwb_controller/src/progress_checker.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dwb_controller/progress_checker.hpp"
+#include <cmath>
+#include "dwb_core/exceptions.hpp"
+#include "nav2_util/duration_conversions.hpp"
+
+namespace dwb_controller
+{
+static double pose_distance(const geometry_msgs::msg::Pose2D &, const geometry_msgs::msg::Pose2D &);
```

Comment:
Reason to pre-declare this function when it could just be placed up here or in the class as an inline method?

Reply:
I was planning to refactor this out to nav2_util at some point in the future, since similar functionality is used in other places in the stack. That's why I left it as a free function.

I pre-declared it because I felt important functions should go at the top of the file, not this.

Reply:
Ok :+1: